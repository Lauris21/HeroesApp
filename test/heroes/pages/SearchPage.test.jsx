import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SearchPage } from "../../../src/heroes";

const mockUseNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockUseNavigate,
}));

describe("Pruebas en <SearchPage />", () => {
  beforeEach(() => jest.clearAllMocks());

  test("Debe mostrarse correctamente con valores por defecto", () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar a Batman y el input con el valor del query", () => {
    render(
      <MemoryRouter initialEntries={["/search?q=batman"]}>
        <SearchPage />
      </MemoryRouter>
    );

    expect(screen.getByText("Batman")).toBeTruthy();

    expect(screen.getByRole("textbox").value).toBe("batman");

    expect(screen.getByRole("img").src).toContain(
      "/assets/heroes/dc-batman.jpg"
    );
  });

  test('Debe mostrar otro componente si no se encuentra el hero "Batman123"', () => {
    render(
      <MemoryRouter initialEntries={["/search?q=batman123"]}>
        <SearchPage />
      </MemoryRouter>
    );

    expect(screen.getByLabelText("alert")).toBeTruthy();
  });

  test("Debe llamar el navigate a la pantalla nueva", () => {
    const valueSearch = "h";
    render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    const input = screen.getByRole("textbox");
    fireEvent.change(input, { target: { value: valueSearch } });

    const form = screen.getByRole("form");
    fireEvent.submit(form);

    expect(mockUseNavigate).toHaveBeenCalledWith(`?q=${valueSearch}`);
  });
});
