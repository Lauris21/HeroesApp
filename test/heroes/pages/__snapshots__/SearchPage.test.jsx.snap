// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> Debe mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1
    class="mt-3 border-bottom border-secondary-subtle p-2"
  >
    Search
  </h1>
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4
        class="mt-3"
      >
        Searching
      </h4>
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control mt-3"
          name="searchText"
          placeholder="search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-3"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4
        class="mt-3"
      >
        Results
      </h4>
      <div
        class="alert alert-primary mt-3 animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
